flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/ZookeeperRegistry.java

```diff
@@ -55,7 +59,7 @@ public void setJobRunning(JobID jobID) throws IOException {
 		try {
 			String zkPath = runningJobPath + jobID.toString();
 			this.client.newNamespaceAwareEnsurePath(zkPath).ensure(client.getZookeeperClient());
-			this.client.setData().forPath(zkPath);
+			this.client.setData().forPath(zkPath, RUNNING.getBytes());
```

Comment:
String to bytes conversion (and bytes to string) must always explicitly specify the encoding (Charset). Otherwise, there can be mismatches when different machines configure different default Charsets.